OutOfSchool/OutOfSchool.AuthCommon/Services/Password/PasswordGenerator.cs

```diff
@@ -1,65 +1,50 @@
-﻿namespace OutOfSchool.AuthCommon.Services.Password;
+﻿using System.Security.Cryptography;
+
+namespace OutOfSchool.AuthCommon.Services.Password;
 
 public static class PasswordGenerator
 {
     /// <summary>
-    /// Generates a Random Password
-    /// respecting the given strength requirements.
+    /// Generates a Random Password.
     /// </summary>
-    /// <param name="passwordOptions">A valid PasswordOptions object
-    /// containing the password strength requirements.</param>
     /// <returns>A random password.</returns>
-    public static string GenerateRandomPassword(PasswordOptions passwordOptions)
+    public static string GenerateRandomPassword()
     {
-        string[] randomChars = new[]
-        {
-            "ABCDEFGHJKLMNOPQRSTUVWXYZ",    // uppercase
-            "abcdefghijkmnopqrstuvwxyz",    // lowercase
-            "0123456789",                   // digits
-            "!@$?_-",                       // non-alphanumeric
-        };
+        string[] allowedCharSets =
+        [
+            "ABCDEFGHJKLMNOPQRSTUVWXYZ",
+            "abcdefghijkmnopqrstuvwxyz",
+            "0123456789",
+            Constants.ValidationSymbols,
+        ];
 
-        CryptoRandom rand = new CryptoRandom();
+        List<char> password = [];
 
-        List<char> chars = new List<char>();
-
-        if (passwordOptions.RequireUppercase)
+        foreach (var charSet in allowedCharSets)
```

Comment:
(Computational hardness) Here you loose the true "randomness". The algorithm makes first 4 characters to strictly be from different sets. 

Please consider adding these 4 characters in random places after the initiating password in the `while (password.Count < Constants.PasswordMinLength)` loop.

Reply:
Fixed. Thank you!